src/main/scala/shine/cuda/ast/Types.scala

```diff
@@ -0,0 +1,78 @@
+package shine.cuda.ast
+
+import shine.C.AST.BasicType
+import shine.DPIA.Nat
+import shine.DPIA.Types.{MatrixLayout, MatrixLayoutIdentifier}
+
+object Wmma {
+  def toString(layout: MatrixLayout): String = {
+    layout match {
+      case MatrixLayout.Row_Major => "nvcuda::wmma::row_major"
+      case MatrixLayout.Col_Major => "nvcuda::wmma::col_major"
+      case i: MatrixLayoutIdentifier =>
+        if (i.layout.isDefined)
+          toString(i.layout.get)
+        else
+          throw new Exception(s"layout $i not infered!")
+      case _ => throw new Exception("this should not happen")
+    }
+  }
+}
+
+case class WmmaAMatrix(m: Nat,
+                       n: Nat,
+                       k: Nat,
+                       dataType: shine.C.AST.Type,
+                       layout: MatrixLayout)
+  extends BasicType("this should not printed") {
```

Comment:
maybe this should not extend `BasicType`? Otherwise is it possible to write `extends BasicType(s"nvcuda::wmma::fragment<...>")`?

Reply:
Writing `extends BasicType(s"nvcuda::wmma::fragment<...>")` is not possible, because the layout is unknown at this point. But this type could extend something else instead of BasicType.

Reply:
Do you have an idea which typ should be extended instead  of BasicType? The other types (`StructType`, `ArrayType`, `PointerType`, `UnionType` and `VectorType`) are not proper either. So should I add a new FragmentDatatype, which extends `Type` directly?

Reply:
I think that would be reasonable, yes.

Reply:
Ok, I changed the Type. So it `extends` `Type` now.
https://github.com/rise-lang/shine/blob/cbd8601d259a431bc8ede1f22dd1f3448aaa74c1/src/main/scala/shine/cuda/ast/Types.scala#L22-L39

Reply:
Because the FragmentType directly  `extends Type` it was necessary to change the following lines. But it seems not good to use a CUDA specfic FragmentType inside parts of the C-Backend.

https://github.com/rise-lang/shine/blob/cbd8601d259a431bc8ede1f22dd1f3448aaa74c1/src/main/scala/shine/C/primitives/imperative/CFunctionDefinition.scala#L130
https://github.com/rise-lang/shine/blob/2e92c849d732701000459279f470091cc71c1522/src/main/scala/shine/C/AST/Types.scala#L33